geopandas/plotting.py

```diff
@@ -366,7 +367,9 @@ def plot_series(
             # https://github.com/edzer/sp/blob/master/R/mapasp.R
         else:
             ax.set_aspect("equal")
-    else:
+    elif aspect == "equal":
+        ax.set_aspect("equal")
```

Comment:
This is redundant, you can leave it out. `elif aspect is not None` covers this case as well.